src/sympc/protocol/aby3/aby3.py

```diff
@@ -284,7 +301,7 @@ def bit_injection(x: MPCTensor, session: Session, ring_size: int) -> MPCTensor:
         x2_sh = []
         x3_sh = []
 
-        for sh in decompose:
+        for sh in list(map(lambda x: x[0].resolve_pointer_type(), decompose)):
```

Comment:
MAYBE: Could you name `decompose` to `decompose_shares` (verbs are usually reserved for methods and nouns for variables)

Reply:
modified